drivers/gpio/gpio_stm32.c

```diff
@@ -737,12 +738,16 @@ static int gpio_stm32_init(const struct device *dev)
 		},							       \
 		.base = (uint32_t *)__base_addr,				       \
 		.port = __port,						       \
-		.pclken = { .bus = __bus, .enr = __cenr }		       \
+		COND_CODE_1(DT_NODE_HAS_PROP(__node, clocks),		       \
+			   (.pclken = { .bus = __bus, .enr = __cenr },),       \
+			   (/* Nothing if clocks not present */))	       \
 	};								       \
 	static struct gpio_stm32_data gpio_stm32_data_## __suffix;	       \
 	PM_DEVICE_DT_DEFINE(__node, gpio_stm32_pm_action);		       \
 	DEVICE_DT_DEFINE(__node,					       \
-			    gpio_stm32_init,				       \
+			    COND_CODE_1(DT_NODE_HAS_PROP(__node, clocks),      \
+			    (gpio_stm32_init),				       \
+			    (NULL)),					       \
```

Comment:
Readability:
```suggestion
			    COND_CODE_1(DT_NODE_HAS_PROP(__node, clocks),      \
					(gpio_stm32_init), (NULL)),	       \
```
(At least indentation is required, use of one line is as you prefer)

Reply:
I think it better to leave each of the results of the `COND_COND` in a separate line, in a `if else endif` fashion.